.github/workflows/generate-js.yml

```diff
@@ -0,0 +1,50 @@
+name: 'Generate JavaScript code and open pull request with it'
+
+on:
+  push:
+    branches-ignore:
+     - 'renovate/*'
+     - 'github-actions/generate-js/*'
+    paths-ignore:
+     - '.github/**'
+     - '**.md'
+     - 'action.yml'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: npm
+
+      - run: npm install
+
+      - run: npm run prepare
+
+      - name: Get word diff to see changes in dist/index.js
+        id: diff
+        run: |
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "GIT_DIFF<<$EOF" >> $GITHUB_OUTPUT
+          git diff --word-diff=porcelain --word-diff-regex=. dist/ | grep '^[+-]' >> $GITHUB_OUTPUT
+          echo "$EOF" >> $GITHUB_OUTPUT
+
+      - name: Create pull request with generated JavaScript
+        uses: peter-evans/create-pull-request@v5
+        with:
+          add-paths: dist/
+          commit-message: Generate JavaScript based on ${{ github.sha }}
+          author: GitHub <noreply@github.com>
+          branch: github-actions/generate-js/${{ github.ref_name }}
+          title: Generate JavaScript for branch `${{ github.ref_name }}`
+          body: |
+            This pull pequest is opened automatically and contains generated JavaScript code based on commit ${{ github.sha }}. The changes are the following (context excluded):
+            ```
+            ${{ steps.diff.outputs.GIT_DIFF }}
```

Comment:
Seems redundant to have in a PR which already has a diff.

Reply:
Explained in the comment above